server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -295,8 +431,37 @@ void collectRemoteException(Exception ex, String clusterAlias, String[] remoteIn
             }
         }
 
-        int size() {
-            return this.indexFailures.size();
+        boolean isEmpty() {
+            return failuresByIndex.isEmpty();
+        }
+    }
+
+    private class NodeTransportHandler implements TransportRequestHandler<FieldCapabilitiesNodeRequest> {
+        @Override
+        public void messageReceived(final FieldCapabilitiesNodeRequest request,
+                                    final TransportChannel channel,
+                                    Task task) throws Exception {
```

Comment:
Should we fork the execution outside of the network thread ? The list of shards can be large on some tiers.

Reply:
This is a good question! What do you think would be the right thread pool (maybe 'management')?